istioctl/pkg/waypoint/waypoint.go

```diff
@@ -491,3 +544,42 @@ func getNamespace(kubeClient kube.CLIClient, ns string) (*corev1.Namespace, erro
 	}
 	return nsObj, nil
 }
+
+func errorWithMessage(errMsg string, gwc *gateway.Gateway, err error) error {
+	errorMsg := fmt.Sprintf("%s\t%v/%v", errMsg, gwc.Namespace, gwc.Name)
+	if err != nil {
+		errorMsg += fmt.Sprintf(": %s", err)
+	}
+	return fmt.Errorf(errorMsg)
+}
+
+func printWaypointStatus(ctx cli.Context, w *tabwriter.Writer, kubeClient kube.CLIClient, gw []gateway.Gateway) error {
+	var cond metav1.Condition
+	startTime := time.Now()
+	ticker := time.NewTicker(1 * time.Second)
+	defer ticker.Stop()
+	fmt.Fprintln(w, "NAMESPACE\tNAME\tSTATUS\tTYPE\tREASON\tMESSAGE")
```

Comment:
May be better not to print the namespace column if the namespace flag is specified, similar to kubectl.

Reply:
that's reasonable... good suggestion!

will update